aws_lambda_powertools/utilities/batch/sqs.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+
+"""
+Batch SQS utilities
+"""
+
+from typing import List
+
+import boto3
+
+from aws_lambda_powertools.middleware_factory import lambda_handler_decorator
+
+from .base import BasePartialProcessor
+
+
+class BasePartialSQSProcessor(BasePartialProcessor):
+    def __init__(self):
+        self._client = boto3.client("sqs")
+        self.success_messages: List = []
+        self.fail_messages: List = []
+
+        super().__init__()
+
+    def get_queue_url(self):
+        *_, account_id, queue_name = self.records[0]["eventSourceARN"].split(":")
+        return f"{self._client._endpoint.host}/{account_id}/{queue_name}"
+
+    def get_entries_to_clean(self):
+        return [{"Id": msg["messageId"], "ReceiptHandle": msg["receiptHandle"]} for msg in self.success_messages]
+
+    def _process_record(self, record):
+        try:
+            result = self.handler(record)
+            return self.success_handler(record, result)
+        except Exception as exc:
+            return self.failure_handler(record, exc)
+
+    def _prepare(self):
+        """
+        """
+        self.success_messages.clear()
+        self.fail_messages.clear()
+
+    def _clean(self):
+        """
+        """
+        if not self.fail_messages:
+            return
+
+        queue_url = self.get_queue_url()
+        entries_to_remove = self.get_entries_to_clean()
+
+        return self._client.delete_message_batch(QueueUrl=queue_url, Entries=entries_to_remove)
+
+
+class DefaultPartialSQSProcessor(BasePartialSQSProcessor):
```

Comment:
I think we can simplify the implementation by just having a single class here. Perhaps just rename `BasePartialSQSProcessor` to `DefaultPartialSQSProcessor` instead of having the extra class just to change the name.

Reply:
Yeah, see your point but I think the prefix `Base` is more appealing from a dev point of view. "It just" makes sense to extend a class with such name, IMO, more than something named `DefaultPartial`... I'll stick with `PartialSQSProcessor`, alright ?

Reply:
I think this one might be a matter of taste, I just felt like the class hierarchy was a bit more complex than necessary. `PartialSQSProcessor` sounds good to me!